row.go

```diff
@@ -519,10 +519,10 @@ type reconstructFunc func(reflect.Value, levels, Row) (Row, error)
 
 func reconstructFuncOf(columnIndex int16, node Node) (int16, reconstructFunc) {
 	switch {
-	case node.Optional():
-		return reconstructFuncOfOptional(columnIndex, node)
 	case node.Repeated():
 		return reconstructFuncOfRepeated(columnIndex, node)
+	case node.Optional():
+		return reconstructFuncOfOptional(columnIndex, node)
```

Comment:
Both this change, and the passthru of the parent's `Optional()` and `Repeated()` attributes are necessary for the current test to not panic.  Whether or not this is the desired behavior is up to the team to decide.